### PR TITLE
add: bdt (boring data tool)

### PR DESCRIPTION
**Pull request by matthias-Q**
## Description of changes
Hi, this is my first time adding a package to NixPkgs. This package is a rust tool that lets you query parquet/json/avro files from the CLI.

Please let me know if I missed something that I should.
Cheers,
Matt

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/bdt/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ pkgs ? import <nixpkgs> { }, lib ? pkgs.lib }:
+pkgs.rustPlatform.buildRustPackage rec {
+  pname = "bdt";
+  version = "0.18.0";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "datafusion-contrib";
+    repo = pname;
```
Comment (lolbinarycat) [inline pkgs/applications/misc/bdt/default.nix:8]:
```suggestion
    repo = "bdt";
```

reusing pname here is discouraged

File: pkgs/applications/misc/bdt/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    rev = "5c6730a8e3cd43c7847aef76b499197730cded58";
+    hash = "sha256-gUKsJwbpMPSO+KPqyJRodrRLjUpTh/y6C2xhrgvJFKk=";
+  };
+
+  cargoHash = "sha256-4KrFhchoIB2N89m7HrL0xj2Z+u/6/6Onxa2wIAX18Io=";
+
+  meta = with lib; {
+    description = "boring data tool. A CLI tool to query parquet, json and avro files.";
+    homepage = "https://github.com/datafusion-contrib/bdt";
+    license = licenses.asl20;
```
Comment (lolbinarycat) [inline pkgs/applications/misc/bdt/default.nix:18]:
```suggestion
    license = lib.licenses.asl20;
```

File: pkgs/applications/misc/bdt/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{ pkgs ? import <nixpkgs> { }, lib ? pkgs.lib }:
```
Comment (lolbinarycat) [inline pkgs/applications/misc/bdt/default.nix:1]:
```suggestion
{ lib
, rustPlatform
, fetchFromGitHub
}:
```

`callPackage` will fill these arguments automatically.

**Review by lolbinarycat — CHANGES_REQUESTED** [review]

**Comment by lolbinarycat** [comment]
also, please add `meta.mainProgram = "bdt"`

**Comment by lolbinarycat** [comment]
additionally, `pkgs/applications/misc/bdt/default.nix` should be moved to `pkgs/by-name/bd/bdt/package.nix`

File: pkgs/applications/misc/bdt/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "datafusion-contrib";
+    repo = "bdt";
+    rev = "5c6730a8e3cd43c7847aef76b499197730cded58";
+    hash = "sha256-gUKsJwbpMPSO+KPqyJRodrRLjUpTh/y6C2xhrgvJFKk=";
+  };
+
+  cargoHash = "sha256-4KrFhchoIB2N89m7HrL0xj2Z+u/6/6Onxa2wIAX18Io=";
+
+  meta = {
+    description = "boring data tool. A CLI tool to query parquet, json and avro files.";
```
Comment (lolbinarycat) [inline pkgs/applications/misc/bdt/default.nix:19]:
```suggestion
    description = "boring data tool. A CLI tool to query parquet, json and avro files";
```

description should not end in a period

**Comment by matthias-Q** [comment]
> additionally, `pkgs/applications/misc/bdt/default.nix` should be moved to `pkgs/by-name/bd/bdt/package.nix`

This broke the checks.

**Comment by lolbinarycat** [comment]
> This broke the checks.

that's because you didn't rename it from `default.nix` to `package.nix`

**Comment by matthias-Q** [comment]
Sorry missed that totally.

**Review by lolbinarycat — APPROVED** [review]

**Review by lolbinarycat — APPROVED** [review]
built successfully on x86_64-linux

**Comment by matthias-Q** [comment]
@lolbinarycat Is there anything that is left? Should I change something here?

File: pkgs/by-name/bd/bdt/package.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  cargoHash = "sha256-4KrFhchoIB2N89m7HrL0xj2Z+u/6/6Onxa2wIAX18Io=";
+
+  meta = {
+    description = "boring data tool. A CLI tool to query parquet, json and avro files";
+    homepage = "https://github.com/datafusion-contrib/bdt";
+    license = lib.licenses.asl20;
+    mainProgram = "bdt";
+    maintainers = [];
```
Comment (SuperSandro2000) [inline pkgs/by-name/bd/bdt/package.nix:23]: We really should set this.

Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

**Comment by matthias-Q** [comment]
@SuperSandro2000 I rebased/squashed and edited commit messages. Hopefully they are now according to the contribution guidelines. Sorry for the hassle.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**